### ISSUE TITLE
License "special exception" clarification

### ISSUE THREAD

**Issue by zultron** [synthetos/g2#399]
I'm looking at the Bezier curve velocity planning in `plan_exec.cpp`, and would like to use it in a ROS MoveIt time parameterization plugin.  Most of ros is BSD licensed, and upstream maintainers ask for that for new contributions.  Vanilla GPLv2 probably wouldn't fly, but I'm hoping that the linking exception, which I don't understand, could somehow make the license compatible.  The linking exception doesn't follow the usual GPL linking exceptions, like the GNU Classpath one.

Bottom line:  Can all or parts of `plan_exec.cpp` be included in [MoveIt!][1] while still allowing binaries to be distributed under the BSD 3-clause license?

Thanks!

[1]: https://github.com/ros-planning/moveit

**Comment by giseburt** [comment]
> Can all or parts of plan_exec.cpp be included in MoveIt! while still allowing binaries to be distributed under the BSD 3-clause license?

@aldenhart and I discussed this and as long as proper attribution is maintained we are fine with this usage of the Bezier curve velocity planning code.

We truly appreciate you asking first. Please let us know how it works out!

-Rob
